packages/utils/telemetry-utils/src/test/config.spec.ts

```diff
@@ -0,0 +1,260 @@
+        assert.equal(config.getNumberArray("badNumberArray"), undefined);
+
+        assert.deepEqual(config.getStringArray("stringArray"), ["1", "2", "3"]);
+        assert.equal(config.getStringArray("badStringArray"), undefined);
+
+        assert.deepEqual(config.getBooleanArray("booleanArray"), [true, false, true]);
+        assert.equal(config.getBooleanArray("badBooleanArray"), undefined);
+        assert.equal(config.getBooleanArray("badBooleanArray2"), undefined);
+    });
+
+    it("Void provider", () => {
+        const config = new CachedConfigProvider(inMemoryConfigProvider(undefined));
+        assert.equal(config.getNumber("number"), undefined);
+        assert.equal(config.getNumber("does not exist"), undefined);
+        assert.equal(config.getBoolean("boolean"), undefined);
+    });
+
+    it("Config priority", () => {
+        const settings1 = {
+            number: "1",
+            string: "string1",
+            boolean: "true",
+            featureEnabled: "false",
+        };
+        const settings2 = {
+            number: "2",
+            string: "string2",
+            boolean: "false",
+            number2: "3",
+            featureEnabled: "true",
+        };
+        const settings3 = {
+            number2: "4",
+            number3: "4",
+            featureEnabled: "true",
+        };
+
+        const config1 = new CachedConfigProvider(
+            inMemoryConfigProvider(getMockStore(settings1)),
+            inMemoryConfigProvider(getMockStore(settings1)),
+            inMemoryConfigProvider(getMockStore(settings2)),
+            inMemoryConfigProvider(getMockStore(settings3)),
+            );
+
+        assert.equal(config1.getNumber("number"), 1); // from settings1
+        assert.equal(config1.getString("string"), "string1"); // from settings1
+        assert.equal(config1.getBoolean("boolean"), true); // from settings1
+        assert.equal(config1.getNumber("number2"), 3); // from settings2
+        assert.equal(config1.getNumber("number3"), 4); // from settings3
+        assert.equal(config1.getBoolean("featureEnabled"), false); // from settings1.BreakGlass
+
+        const config2 = new CachedConfigProvider(
+            inMemoryConfigProvider(getMockStore(settings3)),
+            inMemoryConfigProvider(getMockStore(settings2)),
+            inMemoryConfigProvider(getMockStore(settings1)),
+            inMemoryConfigProvider(getMockStore(settings1)),
+        );
+
+        assert.equal(config2.getNumber("number"), 2); // from settings2
+        assert.equal(config2.getString("string"), "string2"); // from settings2
+        assert.equal(config2.getBoolean("boolean"), false); // from settings2
+        assert.equal(config2.getNumber("number2"), 4); // from settings3
+        assert.equal(config2.getNumber("number3"), 4); // from settings3
+        assert.equal(config1.getBoolean("featureEnabled"), false); // from settings1.BreakGlass
+    });
+
+    // #region SettingsProvider
```

Comment:
How do you feel about removing this region altogether? I was thinking this would be useful but only as a guideline for how consumers of `SettingsProvider` are able to use our interfaces but I'm not entirely sure it belongs in the `main` branch